code/datums/components/slime_friends.dm

```diff
@@ -0,0 +1,52 @@
+/datum/component/slime_friends
+	var/timer
+	var/static/colours = list(
+		/datum/slime_type/adamantine,
+		/datum/slime_type/black,
+		/datum/slime_type/blue,
+		/datum/slime_type/bluespace,
+		/datum/slime_type/cerulean,
+		/datum/slime_type/darkblue,
+		/datum/slime_type/darkpurple,
+		/datum/slime_type/gold,
+		/datum/slime_type/green,
+		/datum/slime_type/grey,
+		/datum/slime_type/lightpink,
+		/datum/slime_type/metal,
+		/datum/slime_type/oil,
+		/datum/slime_type/orange,
+		/datum/slime_type/pink,
+		/datum/slime_type/purple,
+		/datum/slime_type/pyrite,
+		/datum/slime_type/rainbow,
+		/datum/slime_type/red,
+		/datum/slime_type/sepia,
+		/datum/slime_type/silver,
+		/datum/slime_type/yellow,
+	)
+
+/datum/component/slime_friends/Initialize(...)
+	. = ..()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	var/mob/living/living_parent = parent
+	living_parent.faction |= FACTION_SLIME
+	timer = addtimer(CALLBACK(src, PROC_REF(make_slime_friend), living_parent), 20 SECONDS)
```

Comment:
Don't start this timer unless the wizard is not in the den.
Use signals to figure out when they have left the den and then start the timer (unless they gain this component while not in the den).